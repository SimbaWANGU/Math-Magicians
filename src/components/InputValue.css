.inputValue {
  text-align: center;
  display: flex;
  width: 100%;
  height: 100%;
}

.inputValue :hover {
  transform: scale(1.1);
}

.inputValue :active {
  transform: scale(0.9);
}

.nameOfValue {
  width: 100%;
  height: auto;
  font-size: 24px;
  font-weight: 700;
  margin: auto;
}
