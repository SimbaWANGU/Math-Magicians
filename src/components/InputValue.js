/* eslint-disable react/destructuring-assignment */
/* eslint-disable no-useless-constructor */
import React from 'react';
import './InputValue.css';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class InputValue extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="inputValue" style={{ backgroundColor: this.props.backgroundColor }}>
        <p className="nameOfValue">{this.props.content}</p>
      </div>
    );
  }
}

InputValue.propTypes = { content: PropTypes.string.isRequired };
InputValue.propTypes = { backgroundColor: PropTypes.string.isRequired };

export default InputValue;
