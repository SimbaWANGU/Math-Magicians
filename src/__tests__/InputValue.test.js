import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import InputValue from '../components/InputValue';

describe('Input component tests', () => {
  const content = '7';
  const backgroundColor = 'blue';
  const handleClick = () => {
    const random = Math.random();
    return random.toString();
  };

  it('Renders as expected', () => {
    const component = renderer.create(
      <InputValue content={content} backgroundColor={backgroundColor} handleClick={handleClick} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('The user should see a single value for a given input', () => {
    const { getByTestId } = render(
      <InputValue content={content} backgroundColor={backgroundColor} handleClick={handleClick} />,
    );
    expect(getByTestId('calcValue')).toHaveTextContent(content);
  });
});
