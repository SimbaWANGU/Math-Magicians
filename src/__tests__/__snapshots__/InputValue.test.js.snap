// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests Renders as expected 1`] = `
<div
  className="inputValue"
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
>
  <p
    className="nameOfValue"
    data-testid="calcValue"
    onClick={[Function]}
  >
    7
  </p>
</div>
`;
